src/main/java/com/google/api/codegen/transformer/DynamicLangApiMethodTransformer.java

```diff
@@ -103,9 +103,12 @@ public OptionalArrayMethodView generateMethod(
     apiMethod.requestVariableName(namer.getRequestVariableName(method));
     apiMethod.requestTypeName(
         namer.getRequestTypeName(context.getTypeTable(), context.getMethod().getInputType()));
+    apiMethod.responseTypeName(
+        namer.getRequestTypeName(context.getTypeTable(), context.getMethod().getOutputType()));
```

Comment:
What about just `getTypeName`?

Reply:
Hmm, my instinct is that we want something more descriptive, but I don't know what. `getProtoElementTypeName`? `getTypeRefTypeName`? I don't actually think that I prefer either of those to just `getTypeName`...

Reply:
I was able to find an alternative route which allows us to leave the name as is :),

```java
apiMethod.responseTypeName(
        namer.getMessageTypeName(
            context.getTypeTable(), context.getMethod().getOutputType().getMessageType()));
```

Reply:
Looks like I jumped the gun. That doesn't add the classes to our imports, so it won't work.

Reply:
What about renaming `getRequestTypeName` to an overloaded `getMessageTypeName` which accepts a `TypeRef`?

Reply:
I think the reason not to do an overload is because the method supports types other than message types, such as primitives like string.

I think maybe `getAndSaveTypeName` (the `AndSave` because that is the part that the `getMessageTypeName` function you tried out isn't doing).

But I don't feel strongly - go with your best judgement :-)

Reply:
That sounds nice! Updated